src/main/java/org/jboss/remoting3/remote/ClientConnectionOpenListener.java

```diff
@@ -475,10 +475,65 @@ public void handleEvent(final ConduitStreamSourceChannel channel) {
                                 } else {
                                     ProtocolUtils.writeString(sendBuffer, mechanismName);
                                     if (response != null) {
-                                        sendBuffer.put(response);
+                                        // REM3-370: if message does not fit the buffer
+                                        if (sendBuffer.remaining() < response.length) {
+                                            ByteBuffer buffer = ByteBuffer.allocate(sendBuffer.position() + response.length);
```

Comment:
I would say:
ByteBuffer buffer = ByteBuffer.allocate(sendBuffer.position() + response.length);
should be:
ByteBuffer buffer = ByteBuffer.allocate(sendBuffer.remaining() + response.length);

Reply:
@ropalka  nope, that line is correct.
By doing ByteBuffer.allocate(sendBuffer.position() + response.length()) I'm getting a buffer whose size fits everything that was put into sendBuffer + the entire response that must be sent together.
If I did as you suggest, I would be allocating a buffer whose size has no relationship to the length of bytes contained in sendBuffer. Hence, those bytes could or could not fit into the buffer, depending whether sendBuffer.remaining() were bigger than sendBuffer.position() or not.